### PR TITLE
PPF-104 Create project inside insightly

### PR DESCRIPTION
**Pull request by lucwollants**
### Added

- Added Insightly integration to create project

---
Ticket: https://jira.uitdatabank.be/browse/PPF-104

### CODE REVIEW

File: app/Insightly/Objects/ProjectStage.php [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+declare(strict_types=1);
+
+namespace App\Insightly\Objects;
+
+enum ProjectStage: string
+{
+    case TEST = 'test'; // Kick-off gesprek
+    case START = 'start'; // Start bouw (ontwikkeling)
+    case EVALUATION = 'evaluation'; // Evaluatie/kwaliteitscontrole
+    case LIVE = 'live'; // Live en recrring
```
Comment (LucWollants) [inline app/Insightly/Objects/ProjectStage.php:12]:
```suggestion
    case LIVE = 'live'; // Live en recurring
```

File: tests/Insightly/InsightlyClientTest.php [hunk 2]
```diff
@@ -72,6 +72,24 @@ public function test_it_can_create_an_opportunity(): void
...
+            'Test Integration description',
+            Uuid::uuid4(),
+            IntegrationStatus::Draft,
+            []
+        );
+
+        $insightlyId = $this->insightlyClient->projects()->create($integration);
+        $this->assertNotNull($insightlyId);
+
+        $this->insightlyClient->opportunities()->delete($insightlyId);
```
Comment (LucWollants) [inline tests/Insightly/InsightlyClientTest.php:90]:
```suggestion
        $this->insightlyClient->projects()->delete($insightlyId);
```

**Review by LucWollants — COMMENTED** [review]
I did a full review and found 1 issue and 1 typo. If these are fixed the pull request can be merged. But I am not able to approve it because I am the creator.

**Review by bertramakers — APPROVED** [review]

File: app/Insightly/Resources/InsightlyProjectResource.php [hunk 3]
```diff
@@ -19,14 +23,48 @@ public function __construct(
 
     public function create(Integration $integration): int
     {
-        return 0;
+        $request = new Request(
+            'POST',
+            $this->path,
+            [],
+            JSON::encode(
+                (new ProjectSerializer($this->insightlyClient->getPipelines()))
```
Comment (steftrenson) [inline app/Insightly/Resources/InsightlyProjectResource.php:31]: Would it make sense to move this project serializer construction to a method? Feels like there's too much knowledge required here on how to construct such a serializer, knowledge that's repeated further down as well.

Comment (steftrenson) [reply app/Insightly/Resources/InsightlyProjectResource.php:31]: It could even be injected at construction time I think.

**Review by steftrenson — APPROVED** [review]

File: app/Insightly/Resources/InsightlyProjectResource.php (hunk 3 above)
Comment (LucWollants) [reply app/Insightly/Resources/InsightlyProjectResource.php:31]: We could do that, although this is all Insightly-related code.

**Pull request merged**